### PR TITLE
Responsive fix and some other minor changes

### PR DESCRIPTION
**Pull request by coder-Aayush**
**Some fix are:-** 
- change to dart naming convention (for best practice)
- Responsive card added 
- possible to Navigate back to home screen.
-  fix some warnings

### CODE REVIEW

**Comment by ashiishme** [comment]
@sachinmaharjan182  Can you please verify this PR?

File: lib/ui/food_detail.dart [hunk 1]
```diff
@@ -12,53 +12,53 @@ class FoodDetail extends StatelessWidget {
...
+        'Overview',
+        style: GoogleFonts.poppins(
+            textStyle:
+                TextStyle(fontWeight: FontWeight.w400, fontSize: 16.0)),
+                ),
+                verticalMargin(8.0),
+                Expanded(
+        child: Text(
+          'Lorem Ipsum is simply dupesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since thes, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised the release of Letraset sheets containing Lorem Ipsum passages, and more recenp publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
+          style: GoogleFonts.poppins(
```
Comment (cimplesid) [inline lib/ui/food_detail.dart]: I saw that the same font is used over and again why don't you make it global from the theme in MaterialApp. @coder-Aayush

Comment (coder-Aayush) [reply lib/ui/food_detail.dart]: @cimplesid ok sure. Can you check the is the card on homepage is responsive or not? @cimplesid

File: lib/ui/food_detail.dart [hunk 2]
```diff
@@ -5,88 +5,88 @@ import 'package:font_awesome_flutter/font_awesome_flutter.dart';
 import 'package:google_fonts/google_fonts.dart';
 
 class FoodDetail extends StatelessWidget {
-
   final colorRed = Colors.redAccent.withOpacity(0.7);
```
Comment (cimplesid) [inline lib/ui/food_detail.dart:7]: if you are tryna make better name convention use _varName format in class valiables that are supposed to be private. same for functions i wont comment same thing anywhere else

File: lib/ui/food_detail.dart [hunk 3]
```diff
@@ -5,88 +5,88 @@ import 'package:font_awesome_flutter/font_awesome_flutter.dart';
...
     );
   }
 
-  buildProductImage() {
+  buildProductImage(BuildContext context) {
     return Stack(
       children: <Widget>[
         Container(
           padding: const EdgeInsets.all(8.0),
           height: 340,
```
Comment (cimplesid) [inline lib/ui/food_detail.dart:66]: you have access of context in here you could have used device's size as constraints

File: lib/ui/home_page.dart [hunk 4]
```diff
@@ -63,14 +59,14 @@ class _HomePageState extends State<HomePage> {
...
-                        verticalMargin(MARGIN_MEDIUM),
+                        verticalMargin(marginMedium),
                         buildSearchBarWidget(),
                         verticalMargin(100.0),
                         Stack(
                           overflow: Overflow.visible,
                           children: <Widget>[
                             buildPopularFood(),
-//                            buildTopFood(),
+                            // buildTopFood(),
```
Comment (cimplesid) [inline lib/ui/home_page.dart]: remove the comment

File: lib/widgets/custom_text.dart [hunk 5]
```diff
@@ -3,26 +3,26 @@ import 'package:flutter_challenge/utils/widget_utils.dart';
...
 class CustomText extends StatelessWidget {
-  IconData prefixIcon;
-  String text;
-  Color iconColor;
+  final IconData prefixIcon;
+  final String text;
+  final Color iconColor;
 
-  CustomText({this.prefixIcon,this.text,this.iconColor});
+  CustomText({this.prefixIcon, this.text, this.iconColor});
```
Comment (cimplesid) [inline lib/widgets/custom_text.dart]: make prefixIcon and text required and primary params

**Review by cimplesid — CHANGES_REQUESTED** [review]
Check the comments applicable for all files

**Review by ashiishme — CHANGES_REQUESTED** [review]
@coder-Aayush Please make necessary changes pointed by @cimplesid.

**Review by coder-Aayush — COMMENTED** [review]
changes pointed by @cimplesid at #1

**Review by cimplesid — APPROVED** [review]

**Pull request merged**